process/smartContract/builtInFunctions/esdtTransfer.go

```diff
@@ -265,12 +265,26 @@ func checkFrozeAndPause(
 	return nil
 }
 
+func arePropertiesEmpty(properties []byte) bool {
+	for _, property := range properties {
+		if property != 0 {
+			return false
+		}
+	}
+	return true
+}
+
 func saveESDTData(
 	userAcnt state.UserAccountHandler,
 	esdtData *esdt.ESDigitalToken,
 	key []byte,
 	marshalizer marshal.Marshalizer,
 ) error {
+	isValueZero := esdtData.Value.Cmp(zero) == 0
+	if isValueZero && arePropertiesEmpty(esdtData.Properties) {
+		return userAcnt.DataTrieTracker().SaveKeyValue(key, nil)
```

Comment:
will this remove the key also? or the key remains but with `null` value?

Reply:
this removes the key.